File: examples/scala/src/main/scala/com/xebia/functional/xef/scala/auto/Colors.scala
package com.xebia.functional.xef.scala.conversation

import com.xebia.functional.xef.scala.conversation.*
import io.circe.Decoder
import com.xebia.functional.xef.prompt.Prompt

private final case class Colors(colors: List[String]) derives SerialDescriptor, Decoder

@main def runColors: Unit =
  conversation {
    val colors = prompt[Colors](Prompt("A selection of 10 beautiful colors that go well together"))
    println(colors)
  }

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/Colors.kt
package com.xebia.functional.xef.auto

import com.xebia.functional.xef.auto.llm.openai.OpenAI
import com.xebia.functional.xef.auto.llm.openai.prompt
import kotlinx.serialization.Serializable

@Serializable data class Colors(val colors: List<String>)

suspend fun main() {
  OpenAI.conversation {
    val colors: Colors = prompt("a selection of 10 beautiful colors that go well together")
    println(colors)
  }
}
